components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,780 @@
+                    return false;
+                }
+                if(dayOfMonth.length === 1){
+                    errorMsg = "Error : Invalid Cron Expression";
+                    return false;
+                } else {
+                    errorMsg = "Error : Day of month values must be between 1 and 31";
+                    return validateRangeOfDay([dayOfMonth], 1, 31);
+                }
+            }
+            if (dayOfMonth.search(constants.REGEX_FOR_DAYOFMONTH) !== -1) {
+                errorMsg = "Error : Invalid characters used. See tooltip for allowed characters.";
+                return false;
+            }
+            var list = dayOfMonth.split(",");
+            return list.every(function (dayOfMonth){
+                if(dayOfMonth.includes('/') && dayOfWeek === '?') {
+                    if(dayOfMonth.endsWith('/')){
+                        errorMsg = "Error : Day of month values should not end with /";
+                    } else {
+                        errorMsg = "Error : Day of month values must be between 1 and 31";
+                    }
+                    var startingDayOfMonthOptionArr = dayOfMonth.split('/');
+                    var isValidElements = (validateRangeOfDay([startingDayOfMonthOptionArr[0]], 1, 31) &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        validateRangeOfDay([startingDayOfMonthOptionArr[1]], 1, 31));
+                    return isValidElements || isValidFirstElem;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+                    if(dayOfMonth.endsWith('-')){
+                        errorMsg = "Error : Day of month values should not end with - ";
+                    } else {
+                        errorMsg = "Error : Day of month values must be between 1 and 31";
+                    }
+                    return validateRangeOfDay(dayOfMonthRangeArr, 1, 31) &&
+                                    (parseInt(dayOfMonthRangeArr[0]) < parseInt(dayOfMonthRangeArr[1]));
+                } else if(dayOfWeek === '?'){
+                    if(isNaN(parseInt(dayOfMonth))){
+                        errorMsg = "Error : Day of month values should not end with , ";
+                    } else {
+                        errorMsg = "Error : Day of month values must be between 1 and 31";
+                    }
+                    return validateRangeOfDay([dayOfMonth], 1, 31) ||
+                        (dayOfMonth === '*' && dayOfWeek !== '*') || (dayOfMonth === '?' && dayOfWeek !== '?');
+                } else {
+                    errorMsg = "Error: Invalid Cron Expression, Expression cannot contain any values except ? ";
+                    return dayOfMonth === '?';
```

Comment:
don't you need to return false here ? since this is a error scenario. Why are you again do dayOfMonth === '?' ??

Reply:
I do this validation, if day of week contain any values then day of month only accept ? .

Reply:
your concept of validation is wrong. we need to have to go through these. Can you schedule a meeting with lasantha and me please